src/java/org/apache/cassandra/tools/nodetool/ForceCompact.java

```diff
@@ -18,26 +18,37 @@
 
 package org.apache.cassandra.tools.nodetool;
 
-import io.airlift.airline.Arguments;
-import io.airlift.airline.Command;
-
 import java.util.ArrayList;
 import java.util.List;
 
 import org.apache.cassandra.tools.NodeProbe;
-import org.apache.cassandra.tools.NodeTool.NodeToolCmd;
+import org.apache.cassandra.tools.nodetool.layout.CassandraUsage;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Parameters;
 
 import static com.google.common.base.Preconditions.checkArgument;
+import static org.apache.cassandra.tools.nodetool.CommandUtils.parsePartitionKeys;
+import static org.apache.cassandra.tools.nodetool.CommandUtils.concatArgs;
 
 @Command(name = "forcecompact", description = "Force a (major) compaction on a table")
-public class ForceCompact extends NodeToolCmd
+public class ForceCompact extends AbstractCommand
 {
-    @Arguments(usage = "[<keyspace> <table> <keys>]", description = "The keyspace, table, and a list of partition keys ignoring the gc_grace_seconds")
+    @CassandraUsage(usage = "[<keyspace> <table> <keys>]", description = "The keyspace, table, and a list of partition keys ignoring the gc_grace_seconds")
     private List<String> args = new ArrayList<>();
 
+    @Parameters(index = "0", description = "The keyspace name to compact", arity = "0..1")
+    private String keyspace;
+
+    @Parameters(index = "1", description = "The table name to compact", arity = "0..1")
+    private String table;
+
+    @Parameters(index = "2..*", description = "The partition keys to compact", arity = "0..1")
```

Comment:
why do we use arity = "0..1" if multiple partition keys are supported?

Reply:
Fixed.